advanced_samples/CSV/dotnet/CSVtoOCS_pkceTest/OpenTestBrowser.cs

```diff
@@ -0,0 +1,43 @@
+﻿using CSVtoOCS;
+using OpenQA.Selenium;
+using OpenQA.Selenium.Chrome;
+using System;
+using System.Threading;
+
+namespace CSVtoOCS_pkceTest
+{
+    public class OpenTestBrowser : IOpenBrowser
+    {
+        public void OpenBrowser(string url, string userName, string password)
+        {
+            // Automatic login works against Microsoft personal account option only
+            // Must use Live account email that isn't also an AAD account
+            // Account must have no 2FA enabled and the login flow must not have any other additional prompts after password entry
+            ChromeOptions options = new ChromeOptions();
+            options.AddArgument("--allow-insecure-localhost");
```

Comment:
If you also do `options.AddArgument("--headless")`, you should be able to run the test pipeline on `ubuntu-latest`.